examples/darwin-framework-tool/commands/common/CertificateIssuer.mm

```diff
@@ -0,0 +1,162 @@
+        error = [NSError errorWithDomain:@"Error" code:0 userInfo:@{ @"reason" : @"nextNodeID is nil" }];
+        completion(nil, error);
+        return;
+    }
+
+    __auto_type * csr = csrInfo.csr;
+    __auto_type * rawPublicKey = [MTRCertificates publicKeyFromCSR:csr error:&error];
+    if (error != nil) {
+        completion(nil, error);
+        return;
+    }
+
+    NSDictionary * attributes = @{
+        (__bridge NSString *) kSecAttrKeyType : (__bridge NSString *) kSecAttrKeyTypeECSECPrimeRandom,
+        (__bridge NSString *) kSecAttrKeyClass : (__bridge NSString *) kSecAttrKeyClassPublic
+    };
+    CFErrorRef keyCreationError = NULL;
+    SecKeyRef publicKey
+        = SecKeyCreateWithData((__bridge CFDataRef) rawPublicKey, (__bridge CFDictionaryRef) attributes, &keyCreationError);
+
+    __auto_type * operationalCert = [self issueOperationalCertificateForNodeID:self.nextNodeID
+                                                                      fabricID:self.fabricID
+                                                                     publicKey:publicKey
+                                                                         error:&error];
+
+    // Release no-longer-needed key before we do anything else.
+    CFRelease(publicKey);
+
+    if (error != nil) {
+        completion(nil, error);
+        return;
+    }
+
+    __auto_type * certChain = [[MTROperationalCertificateChain alloc] initWithOperationalCertificate:operationalCert
+                                                                             intermediateCertificate:nil
+                                                                                     rootCertificate:self.rootCertificate
+                                                                                        adminSubject:nil];
+    completion(certChain, nil);
+}
+
+- (MTRCertificateDERBytes _Nullable)issueOperationalCertificateForNodeID:(NSNumber *)nodeID
+                                                                fabricID:(NSNumber *)fabricID
+                                                               publicKey:(SecKeyRef)publicKey
+                                                                   error:(NSError * __autoreleasing *)error
+{
+    __auto_type * operationalCert = [MTRCertificates createOperationalCertificate:self.signingKey
+                                                               signingCertificate:self.rootCertificate
+                                                             operationalPublicKey:publicKey
+                                                                         fabricID:fabricID
+                                                                           nodeID:nodeID
+                                                            caseAuthenticatedTags:nil
+                                                                            error:error];
+    if (error && *error != nil) {
+        return nil;
+    }
+
```

Comment:
```suggestion
```

Returning non-nil but setting error would be a very wrong behavior for the callee....